### PR TITLE
PWN-8734 - Striga. KYC debug settings banner mock

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8734

## Description of Work

- mock KYC banner in debug settings

### CODE REVIEW

**Review by Davran331122 — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsAdapter.kt [hunk 1]
```diff
@@ -56,6 +56,7 @@ class DebugSettingsAdapter(
             is SettingsRow.Info -> R.layout.item_settings_info
             is SettingsRow.Logout -> R.layout.item_settings_logout
             is SettingsRow.Switcher -> R.layout.item_settings_debug_switch
+            else -> error("")
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsAdapter.kt:59]: give some good message, even for debug purposes

File: app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsContract.kt [hunk 2]
```diff
@@ -8,13 +8,14 @@ import org.p2p.wallet.settings.model.SettingsRow
...
-        fun showSettings(item: List<SettingsRow>)
+        fun showSettings(items: List<SettingsRow>)
     }
 
     interface Presenter : MvpPresenter<View> {
         fun loadData()
         fun onNetworkChanged(newNetworkEnvironment: NetworkEnvironment)
         fun switchMoonpayUrl(isSandboxSelected: Boolean)
         fun switchNameServiceUrl(isProdSelected: Boolean)
+        fun onSettingsPopupMenuClicked(s: String)
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsContract.kt:19]: same here

**Review by 1sakov — APPROVED** [review]
lgtm

File: app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsAdapter.kt (hunk 1 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsAdapter.kt:59]: yeah sure, im gonna clean this all up

File: app/src/main/java/org/p2p/wallet/common/feature_toggles/toggles/inapp/InAppFeatureFlag.kt [hunk 3]
```diff
@@ -3,4 +3,9 @@ package org.p2p.wallet.common.feature_toggles.toggles.inapp
 sealed class InAppFeatureFlag {
     abstract val featureName: String
     abstract var featureValue: Boolean
+
+    abstract class InAppFeatureFlagString : InAppFeatureFlag() {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/common/feature_toggles/toggles/inapp/InAppFeatureFlag.kt:7]: not a good solution, we need another for sure

File: app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsFragment.kt [hunk 4]
```diff
@@ -31,19 +38,26 @@ class DebugSettingsFragment :
...
         fun create(): DebugSettingsFragment = DebugSettingsFragment()
     }
 
     override val presenter: DebugSettingsContract.Presenter by inject()
 
     private val binding: FragmentDebugSettingsBinding by viewBinding()
-    private val adapter = DebugSettingsAdapter(::onSettingsRowClicked, ::onSettingsSwitchClicked)
+
+    private val adapterDelegate = SettingsRowAdapter(
+        settingsRowTitleItemDelegate(),
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsFragment.kt:49]: adapter delegates, yay!

File: app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsPresenter.kt [hunk 5]
```diff
@@ -122,10 +125,23 @@ class DebugSettingsPresenter(
                 titleResId = R.string.debug_settings_logs_title,
                 subtitle = resources.getString(R.string.debug_settings_logs_subtitle),
                 iconRes = R.drawable.ic_settings_cloud
-            )
+            ),
+            SettingsRow.PopupMenu(
+                titleResId = R.string.debug_settings_logs_title,
+                selectedItem = inAppFeatureFlags.strigaKycBannerMockFlag.featureValueString ?: "NO_MOCK",
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/debug/settings/DebugSettingsPresenter.kt:131]: gonna be extracted to const

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**